### PR TITLE
chore(tests): adding more tests for lex package

### PR DESCRIPTION
**Pull request by MaxxReis**
## Problem

Low coverage in the lex package

## Solution

Creating tests for increased test coverage for the lex package.

### CODE REVIEW

**Review by matthewmcneely — DISMISSED** [review]

File: schema/schema_test.go [hunk 1]
```diff
@@ -0,0 +1,1291 @@
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := GetWriteContext(tt.args.ctx); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("GetWriteContext() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_state_DeleteAll(t *testing.T) {
```
Comment (matthewmcneely) [inline schema/schema_test.go:64]: Can you standardize the test function names to camel case (TestFooBar) or the also acceptable mixed (Test_FooBar)?

**Review by matthewmcneely — CHANGES_REQUESTED** [review]

File: schema/schema_test.go (hunk 1 above)
Comment (joshua-goldstein) [reply schema/schema_test.go:64]: +1 this is a Go convention, in our repo we tend to use pure camel case (e.g. TestExampleOne)

File: lex/iri_test.go [hunk 2]
```diff
@@ -71,3 +72,103 @@ func TestIsIRIRefChar(t *testing.T) {
...
+	}
+}
+
+func TestIRIRefErrorf(t *testing.T) {
+	type args struct {
+		l    *Lexer
+		styp ItemType
+		want error
+	}
+	tests := []args{
```
[the remark this answers is not in the document]
Comment (joshua-goldstein) [reply lex/iri_test.go:146]: +1 types can be reused and if possible should be declared outside the test functions themselves if they are being reused, see other areas of our codebase

File: lex/iri_test.go [hunk 3]
```diff
@@ -71,3 +72,103 @@ func TestIsIRIRefChar(t *testing.T) {
...
+		{l: &Lexer{
+			Input:      ">",
+			Start:      0,
+			Pos:        0,
+			Width:      0,
+			widthStack: []*RuneWidth{},
+			items:      []Item{},
+			Depth:      0,
```
[the remark this answers is not in the document]
Comment (joshua-goldstein) [reply lex/iri_test.go:89]: +1 types have a default value (e.g. integers default to 0, strings default to empty string ""), they don't need to be declared explicitly

**Review by joshua-goldstein — CHANGES_REQUESTED** [review]

File: lex/iri_test.go [hunk 4]
```diff
@@ -71,3 +72,103 @@ func TestIsIRIRefChar(t *testing.T) {
 		require.Equal(t, test.expResult, result)
 	}
 }
+
+func TestIRIRef(t *testing.T) {
+	type args struct {
+		l    *Lexer
+		styp ItemType
+	}
+	tests := []args{
```
[the remark this answers is not in the document]
Comment (joshua-goldstein) [reply lex/iri_test.go:81]: +1

**Comment by joshua-goldstein** [comment]
Summary of feedback from our discussion:
- This is a very large PR: focus this PR on lex package, open another PR for schema
- Unit test plugins like [this](https://betterprogramming.pub/a-quick-way-to-generate-go-tests-in-visual-studio-code-b7c675b88dac) will generate unit tests by created the necessary input for a function, running it through the function, and making a test based on this output.  However it has no way of knowing what the programmer really _wanted_ to do in the function, so...
- Create unit tests from the beginning so you fully understand what each test is doing.  Test edge cases and error conditions.  Use other tests in lex package (and other packages) as a model.

Other points:
- Declaring slices/other types in one line as opposed to multiple lines for readability
- Use default null values (e.g. default integer value is 0)
- Reuse type definitions whenever possible, and think carefully before creating new types (i.e. is this new type really necessary?)
